src/js/trove/make-standalone.js

```diff
@@ -41,68 +67,53 @@
-        fs.writeSync(outFile, "var requirejs = require(\"requirejs\");\n");
-        fs.writeSync(outFile, "var define = requirejs.define;\n}\n");
-        var toFetchKeys = Object.keys(filesToFetch);
-        toFetchKeys.sort();
-        toFetchKeys.forEach(function(f) {
-          var contents = fs.readFileSync(filesToFetch[f], {encoding: 'utf8'});
-          fs.writeSync(outFile, contents);
-        });
-        fs.writeSync(outFile, "define(\"program\", " + depsLine + ", function() {\nreturn ");
-        var writeRealOut = function(str) { 
-          fs.writeSync(outFile, str, {encoding: 'utf8'}); 
-          return runtime.nothing; 
-        };
-        return runtime.safeCall(function() { 
-          return runtime.getField(body, "print-ugly-source").app(runtime.makeFunction(writeRealOut, "write-real-out"));
-        }, function(_) {
-          fs.writeSync(outFile, "\n});\n");
-          fs.writeSync(outFile, handalone);
-          fs.fsyncSync(outFile);
-          fs.closeSync(outFile);
-          return true;
-        });
+      if(!config["use-raw-files"]) {
+        throw new Error("Cannot not use raw-files! RequireJS is gone");
       }
-      else {
-        return runtime.pauseStack(function(restarter) {
-          requirejs.optimize(config, function(result) {
-            var programWithDeps = fs.readFileSync(config.out, {encoding: 'utf8'});
-            // Browser/node check based on window below
-            fs.open(realOut, "w", function(err, outFile) {
-              if (err) throw err;
-              fs.writeSync(outFile, "if(typeof window === 'undefined') {\n");
-              fs.writeSync(outFile, "var requirejs = require(\"requirejs\");\n");
-              fs.writeSync(outFile, "var define = requirejs.define;\n}\n");
-              fs.writeSync(outFile, programWithDeps);
-              fs.writeSync(outFile, "define(\"program\", " + depsLine + ", function() {\nreturn ");
-              var writeRealOut = function(str) { 
-                fs.writeSync(outFile, str, {encoding: 'utf8'}); 
-                return runtime.nothing; 
-              };
-              runtime.runThunk(function() { 
-                return runtime.getField(body, "print-ugly-source").app(runtime.makeFunction(writeRealOut, "write-real-out"));
-              }, function(_) {
-                fs.writeSync(outFile, "\n});\n");
-                fs.writeSync(outFile, handalone);
-                fs.fsyncSync(outFile);
-                fs.closeSync(outFile);
-                restarter.resume(true);
-              });
-            });
-          }, function(err) {
-            console.error("Error while using requirejs optimizer: ", err); 
-            restarter.error(runtime.ffi.makeMessageException("Error while using requirejs optimizer: ", String(err)));
-          });
-        });
+      var outFile = fs.openSync(realOut, "w");
+
+      // Write the amd loader first
+      var loaderContents = fs.readFileSync(AMD_LOADER, READ_OPTIONS);
```

Comment:
This should probably be (a) configurable and (b) tested for relative-path-ness depending on where the compiler was run from.